### ISSUE TITLE
Opening pull popup fails unless status was opened prior

### ISSUE THREAD

**Issue by gegoune** [NeogitOrg/neogit#639]
### Description

Continuation of discussion started in https://github.com/NeogitOrg/neogit/issues/637#issuecomment-1640751018.

Users were previously able to call all popups without having opened status buffer before, it has changed in one of the recent commits.

I am loving this minimal reproducer btw!

### Neovim version

NVIM v0.10.0-dev-682+gd0d132fbd-Homebrew
Build type: Release
LuaJIT 2.1.0-beta3

### Operating system and version

macOS

### Steps to reproduce

1. nvim --clean -u minimal.lua

(I have added offender there)

### Expected behavior

Show pull popup without any errors.

### Actual behavior

```
E5113: Error while calling lua chunk: /tmp/nvim_issue/neogit//lua/neogit/popups/pull/init.lua:26: attempt to
 concatenate local 'current' (a nil value)
stack traceback:
        /tmp/nvim_issue/neogit//lua/neogit/popups/pull/init.lua:26: in function 'c'
        /tmp/nvim_issue/neogit//lua/neogit/popups/init.lua:8: in function 'f'
        /tmp/nvim_issue/neogit//lua/neogit/popups/init.lua:17: in function </tmp/nvim_issue/neogit//lua/neog
it/popups/init.lua:11>
        /Users/gegoune/repos/dots/nvim/minimal.init.lua:13: in main chunk
```

### Minimal config

```Lua
for name, url in pairs {
  neogit = 'https://github.com/NeogitOrg/neogit',
  plenary = 'https://github.com/nvim-lua/plenary.nvim',
  telescope = 'https://github.com/nvim-telescope/telescope.nvim',
} do
  local install_path = vim.fn.fnamemodify('/tmp/nvim_issue/' .. name, ':p')
  if vim.fn.isdirectory(install_path) == 0 then vim.fn.system { 'git', 'clone', '--depth=1', url, install_path } end
  vim.opt.runtimepath:append(install_path)
end

require('neogit').setup()
print 'Minimal Config Ready!'
require('neogit.popups').open 'pull'()
```

**Comment by treatybreaker** [comment]
I'll get a git bisect going to determine where this error first began.

**Comment by gegoune** [comment]
And it does for me too, thank you!

**Comment by CKolkey** [comment]
So, that _will_ work, but it wont _work_. I'll explain. It "fixes" the issue of trying to concatonate nil, but doesn't actually solve the problem. I think the better solution is to go through the git lib and use https://github.com/NeogitOrg/neogit/blob/master/lua/neogit/lib/git/branch.lua#L66 instead. This will try to use the internal state _if possible_, and will fall back to the CLI if not.

**Comment by treatybreaker** [comment]
I'll update my commit, see if I can use that. You're right unfortunately :(.

**Comment by CKolkey** [comment]
Haha, yeah, I hate when that happens :P

**Comment by Massolari** [comment]
I still have this issue.
Tried with the minimal config from the issue description:
![image](https://github.com/NeogitOrg/neogit/assets/12272702/d9b74712-f26c-4bd5-b388-3874f5762967)

**Comment by gegoune** [comment]
Has this resurfaced again with slightly different error message this time? I am getting

```lua
E5108: Error executing lua: ...e/pack/pckr/opt/neogit/lua/neogit/popups/commit/init.lua:33: attempt to index local 'env' (a nil value)
stack traceback:
        ...e/pack/pckr/opt/neogit/lua/neogit/popups/commit/init.lua:33: in function 'create'
        ...ocal/share/nvim/site/pack/pckr/opt/neogit/lua/neogit.lua:79: in function 'open'
```

@CKolkey Should I open new issue for it? This is once again, without calling status first. :)

**Comment by CKolkey** [comment]
Right... 😓 I never use that workflow, so, sorry for breaking it again. I started passing a table into the popup constructor for the commit popup, since now fixup/squash/augment/instant-fixup/instant-squash will use the commit under the cursor (if there is one) instead of opening the commit select buffer... bla bla bla, launching the popup via `:Neogit commit` takes a different code path, and no table was passed in. 

This should do the trick: https://github.com/NeogitOrg/neogit/pull/730

**Comment by gegoune** [comment]
Don't worry and thank you very much for rapid fix! I like passed in table, **even** if it's empty! 😀
